### PR TITLE
refac: minimize using template types

### PR DESCRIPTION
**Pull request by chokobole**
# Description

This PR minimizes using template types. Our problem arises when creating c binding APIs for `ProvingKey`. Since the template type of `ProvingKey` is `PCS`, the corresponding APIs for creating this structure should be `tachyon_bn254_shplonk_plonk_proving_key_create()`. However, when looking at the members included, it doesn't actually depend on `PCS`. So the team decided on the following:

**Other than `Entity<PCS>`, DO NOT USE template `PCS`**

### CODE REVIEW

**Review by TomTaehoonKim — CHANGES_REQUESTED** [review]
LGTM

File: tachyon/zk/base/blinder_unittest.cc [hunk 1]
```diff
@@ -56,26 +56,36 @@ class FakeRandomFieldGenerator : public RandomFieldGeneratorBase<math::GF7> {
...
+  for (size_t i = 0; i < 2; ++i) {
+    bool include_last_row = i == 0;
+
+    FakeRandomFieldGenerator generator(blinding_values);
+    Blinder<FakePCS> blinder(&generator, kBlindingFactors);
+
+    RowIndex blinded_rows = kBlindingFactors;
+    if (include_last_row) {
+      ++blinded_rows;
+    }
```
[the remark this answers is not in the document]
Comment (chokobole) [reply tachyon/zk/base/blinder_unittest.cc:53]: Actually this is changed at https://github.com/kroma-network/tachyon/pull/253/files#diff-f2413a4de51629169154be6d4a9fbb49abdce0a5f12d6eed5355cc196708fbb7R38-R39. 

Do you think it's more readable? I think it's styling issue and it is different from people to people, and I should've not modified it there. :(

[the remark this answers is not in the document]
Comment (TomTaehoonKim) [reply tachyon/zk/base/blinder_unittest.cc:53]: Oh yeah, I think it's just preference issue. Then could you just fix like below?
```suggestion
    RowIndex blinding_rows = kBlindingFactors;
    if (include_last_row) ++blinding_rows;
```

[the remark this answers is not in the document]
Comment (chokobole) [reply tachyon/zk/base/blinder_unittest.cc:53]: You want to make it look similar to the `Blind()`, right? Okay

**Review by TomTaehoonKim — APPROVED** [review]
LGTM

**Review by Insun35 — APPROVED** [review]
LGTM

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Pull request merged**